File: 2018/Senior/Senior1.cpp
#include<iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <iomanip>
using namespace std;

int main() {
    int N;
    int input;
    cin >> N;
    vector<float> villages;
    for(int i = 0; i < N; i++) {
        cin >> input;
        villages.push_back(input);
    }   
    sort(villages.begin(), villages.end());
    float Total = 0;
    for(int i = 0; i < villages.size(); i++) {
        float sum1 = villages[i+1] + villages[i]; //25
        float mid1 = sum1/2; //12.5
        float sum2 = villages[i+1] + villages[i+2]; //31
        float mid2 = sum2/2; //15.5
        float temp = mid1 - mid2;
        if(temp < 0) {
            temp = temp * (-1);
        }
        if(temp < Total or Total == 0) {
            Total = temp;
        }
    }
    cout << fixed << setprecision(1);
    cout << Total << endl;
}

File: 2018/Senior/Senior1.py
n = int(input())
distinct = []

for i in range(n):
  distinct.append(float(input()))

distinct = [int(i) for i in distinct]
distinct.sort()

#[16, 0, 1, 2, 15]
#[0, 4, 10, 15, 16]

Total = 0

for i in range(len(distinct)-2):
  sum1 = distinct[i+1] + distinct[i] #25
  mid1 = sum1/2 #12.5
  sum2 = distinct[i+1] + distinct[i+2] #31
  #else:
    #sum2 = distinct[i] + distinct[i-1]
  mid2 = sum2/2 #15.5
  Temp = mid1 - mid2 #-3
  if Temp < 0: 
    Temp = Temp * (-1)#3
  if Temp < Total or Total == 0:
    Total = Temp#3

print(Total)
